presto-hive/src/main/java/com/facebook/presto/hive/HiveTableLayoutHandle.java

```diff
@@ -268,7 +268,7 @@ public Object getIdentifier(Optional<ConnectorSplit> split, PlanCanonicalization
         if (split.isPresent() && (split.get() instanceof HiveSplit) && domainPredicate.getColumnDomains().isPresent()) {
             HiveSplit hiveSplit = (HiveSplit) split.get();
             Set<Subfield> subfields = hiveSplit.getRedundantColumnDomains().stream()
-                    .map(column -> new Subfield(((HiveColumnHandle) column).getName()))
+                    .map(column -> new Subfield(column.getName()))
```

Comment:
Like the refactoring (replacing with HiveColumnHandle). Thank you for making the codebase better.